modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/enricher/AbstractAnnotationBasedParameterModelEnricher.java

```diff
@@ -0,0 +1,82 @@
+import org.mule.extension.api.introspection.declaration.fluent.ParameterDeclaration;
+import org.mule.module.extension.internal.model.AbstractAnnotatedModelEnricher;
+import org.mule.module.extension.internal.model.property.DeclaringMemberModelProperty;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * A base class to be used to enrich a {@link ParameterDeclaration} when the associated {@link DeclaringMemberModelProperty} has a field that was annotated with a specific {@link Annotation}.
+ *
+ * @since 4.0
+ */
+public abstract class AbstractAnnotationBasedParameterModelEnricher extends AbstractAnnotatedModelEnricher
+{
+
+    @Override
+    public void enrich(DescribingContext describingContext)
+    {
+        DeclarationDescriptor descriptor = describingContext.getDeclarationDescriptor();
+        Declaration declaration = descriptor.getRootDeclaration().getDeclaration();
+
+        //TODO MULE-9083 Refactor to use a Parameterizable interface to make this polymorphic
+        declaration.getConfigurations().forEach(parameterProvider ->
+                                                        enrichParameters(() -> parameterProvider.getParameters()));
+
+        declaration.getConnectionProviders().forEach(parameterProvider ->
+                                                             enrichParameters(() -> parameterProvider.getParameters()));
+        declaration.getOperations().forEach(parameterProvider ->
+                                                    enrichParameters(() -> parameterProvider.getParameters()));
+    }
+
+    private void enrichParameters(Supplier<List<ParameterDeclaration>> parameters)
+    {
+        parameters.get().forEach(this::addModelPropertyIfRequired);
+    }
+
+    /**
+     * Adds the property model provided by {@link #getModelProperty()} if the parameter declaration has a {@link DeclaringMemberModelProperty} , and it contains a field annotated with the annotation provided by {@link #getAnnotationClass()}.
+     */
+    private void addModelPropertyIfRequired(ParameterDeclaration parameterDeclaration)
+    {
+        DeclaringMemberModelProperty declaringMember = parameterDeclaration.getModelProperty(DeclaringMemberModelProperty.KEY);
+        if (declaringMember != null)
+        {
+            Field field = declaringMember.getDeclaringField();
+            if (field.isAnnotationPresent(getAnnotationClass()))
+            {
+                parameterDeclaration.addModelProperty(getModelPropertyKey(), getModelProperty());
+            }
+        }
+    }
+
+    /**
+     * Gets the annotation that this model enricher will look for.
+     */
+    protected abstract Class<? extends Annotation> getAnnotationClass();
+
+    /**
+     * Gets the key that will be used to populate a model property
+     */
+    protected abstract String getModelPropertyKey();
```

Comment:
this method should received the parameterDeclaration, even if you don't use it for your cases